packages/validator/src/validate.decorator.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/validator
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {JSONSchema6} from 'json-schema';
+import * as AJV from 'ajv';
+import {
+  MetadataAccessor,
+  MetadataInspector,
+  ParameterDecoratorFactory,
+} from '@loopback/metadata';
+import * as HttpErrors from 'http-errors';
+
+const debug = require('debug')('loopback:validator');
+
+// tslint:disable-next-line:no-any
+export type Validator = (arg: any) => boolean | Promise<boolean>;
+
+export const VALIDATION_KEY = MetadataAccessor.create<JSONSchema6 | Validator>(
+  'validation.parameter',
+);
+
+export function validatable() {
+  return function(
+    target: Object,
+    member: string,
+    // tslint:disable-next-line:no-any
+    descriptor: TypedPropertyDescriptor<(...args: any[]) => any>,
+  ) {
+    const originalMethod = descriptor.value;
+    if (!originalMethod) {
+      throw new Error(
+        'No method found for some reason, this should not happen',
+      );
+    }
+    // tslint:disable-next-line:no-any
+    descriptor.value = async function(...args: any[]) {
+      const ajv = new AJV();
```

Comment:
+1 for using AJV for validations.

To get the best performance, we need to cache compiled validators and reuse them across requests. Please consider this requirement in your design. See https://www.npmjs.com/package/ajv#performance and https://www.npmjs.com/package/ajv#compileobject-schema---functionobject-data.